### ISSUE TITLE
Error while running Magetbrain using Docker

### ISSUE THREAD

**Issue by neer86upadhyay** [CoBrALab/MAGeTDocker#7]
Dear Gabriel,
We are trying to segment the Habenula from the T1 images acquired from a subjects underwent a longitudinal intervention using the Magetbrain. we already pulled the docker container and arranged the data as per instructions. The commands we ran are as follows:

Start Magetbrain in docker
```
docker run -it -v `pwd`:`pwd` -v '/home/fmrilab/Dokumente/Studien/RUNSTUD/':'/home/fmrilab/Dokumente/Studien/RUNSTUD/' -w `pwd` gdevenyi/magetdocker /bin/bash

import atlases
mb import /home/fmrilab/Dokumente/Studien/RUNSTUD/brains_t1/brain1_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/atlases/Habenula/brain1_labels.mnc

mb import /home/fmrilab/Dokumente/Studien/RUNSTUD/brains_t1/brain2_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/atlases/Habenula/brain2_labels.mnc

mb import /home/fmrilab/Dokumente/Studien/RUNSTUD/brains_t1/brain3_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/atlases/Habenula/brain3_labels.mnc

mb import /home/fmrilab/Dokumente/Studien/RUNSTUD/brains_t1/brain4_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/atlases/Habenula/brain4_labels.mnc

mb import /home/fmrilab/Dokumente/Studien/RUNSTUD/brains_t1/brain5_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/atlases/Habenula/brain5_labels.mnc


convert niftis to mnc (command nii2mnc)


check
mb status

run analysis
mb run

run analysis in parallel

 mb run -q parallel

Please have a look at the following error and help us in sorting this out.

mb_register_labelmask_antsReg_allANTs /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/04LIUS27_T6_T1w_biascorr_brain.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/04LIUS27_T6_T1w_biascorr_brain/nl.xfm ValueError: could not convert string to float: ' cant read /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc'
mb_register_labelmask_antsReg_allANTs /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/04LIUS27_T6_T1w_biascorr_brain.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/04LIUS27_T6_T1w_biascorr_brain/nl.xfm + moving_minimum_resolution=
qbatch: error: local run call returned error code 101
Traceback (most recent call last):
File "/opt/MAGeTbrain/bin/mb", line 792, in 
main()
File "/opt/MAGeTbrain/bin/mb", line 220, in main
options.func(options)
File "/opt/MAGeTbrain/bin/mb", line 329, in command_run
p.run()
File "/opt/MAGeTbrain/bin/mb", line 608, in run
self.queue.run(stages=self._get_stage_order())
File "/opt/MAGeTbrain/bin/mb", line 539, in run
processors=processors)
File "/opt/MAGeTbrain/bin/mb", line 559, in qbatch
opt_afterok, opt_batchsize, opt_walltime), input='\n'.join(commands))
File "/opt/MAGeTbrain/bin/mb", line 492, in execute
raise Exception("Returns %i :: %s" % (proc.returncode, command))
Exception: Returns 1 :: qbatch --jobname mb_templatelib_2021-05-03T15-21-04 --depend NONE* --chunksize 8 --walltime 6:00:00 -
```

Best regards,
Neeraj

**Comment by gdevenyi** [comment]
> convert string to float: ' cant read /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc'

Looks like you're having some file issues. Please check (within th econtainer) if you can access the files properly (something like `mincinfo`

**Comment by gdevenyi** [comment]
Should also try `PrintHeader` since thats the actual command that seems to be having issues.

**Comment by neer86upadhyay** [comment]
Dear Gabriel,
 I tried running both commands but I only got the following output for the
mincinfo while other command didn't even run.
***@***.***:/home/fmrilab# mincinfo
'/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/subjects/brains/03ERIM05_T0_T1w_biascorr_brain.mnc'

file:
/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/subjects/brains/03ERIM05_T0_T1w_biascorr_brain.mnc
image: signed__ float 0 to 536.51763916015625
image dimensions: zspace yspace xspace
    dimension name         length         step        start
    --------------         ------         ----        -----
    zspace                    170           -1      85.8248
    yspace                    256           -1      127.822
    xspace                    192           -1      95.5192
***@***.***:/home/fmrilab# mincinfo
'/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/subjects/brains/03ERIM05_T2_T1w_biascorr_brain.mnc'

file:
/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/subjects/brains/03ERIM05_T2_T1w_biascorr_brain.mnc
image: signed__ float 0 to 577.64923095703125
image dimensions: zspace yspace xspace
    dimension name         length         step        start
    --------------         ------         ----        -----
    zspace                    170           -1       86.228
    yspace                    256           -1      127.722
    xspace                    192           -1      95.1167

Could you please suggest any other way to overcome the issue.

Best Regards,
Neeraj upadhyay
Research Associate
Clinic for Diagnostics and interventional Radiology,
Uniklinikum Bonn

https://www.ukbonn.org/radiologie/forschung/klinische-funktionelle-neurobildgebung/
https://www.ukbonn.org/radiologie/unser-team/wissenschaftliche-mitarbeiter/

**Comment by gdevenyi** [comment]
Looks to me like the files you have provided are corrupted. Please re-collect the originals.

**Comment by gdevenyi** [comment]
You don't seem to have provided any `mincinfo` outputs.

**Comment by neer86upadhyay** [comment]
Dear Gabriel, 
 thanks for the suggestion. after downloading the new files. I was able to ran the mb run -q parallel command but it stopped in between without creating any output file in the subfolders maned intermediate and fusion.
 
 when I checked the whole run it was showing some points like :  file NOMASK does not exist 

after it stopped I checked by running the mb status command which showed the following output:

5 atlases, 21 templates, 111 subjects found
0 atlas-to-template registration commands left
2331 template-to-subject registration commands left
11655 transform merging commands left
11655 label propagation commands left
111 label fusion commands left

later when I tried running the mb run -q parallel  it showed the following report:
5 atlases, 21 templates, 111 subjects found
'vanilla' mode only works for the 'qbatch' queue.
Other queues treat this as stage as 'register'. You will need to run the 'voting' stage separately.

Could please suggest what may be the reason and How I can resolve it. 

Best Regards, 
Neeraj

**Comment by gdevenyi** [comment]
> but it stopped in between without creating any output file in the subfolders named intermediate and fusion.

You haven't provided any logs or outputs which can enable me to help at all.

Based on what you said, you have not read the following warning that printed:
```
            warning("'vanilla' mode only works for the 'qbatch' queue.")
            warning("Other queues treat this as stage as 'register'. " +
                    "You will need to run the 'voting' stage separately.")
```

Therefore, what is likely is, you haven't run the full pipeline.

**Comment by neer86upadhyay** [comment]
Dear Gabriel, 

As such there was no log file in the output but I am copying the log from the terminal for one subject.
 The Log of mb run -q parallel command  is as follows:

++ mktemp -d
+ tmpdir=/tmp/tmp.CyxYrpgl3K
+ atlas=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc
+ target=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc
+ output_xfm=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/nl.xfm
++ basename /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc .mnc
+ atlas_stem=brain1_t1
++ basename /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc .mnc
+ target_stem=07ELAN15_T0_T1w_biascorr_brain
+++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc
++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains
+ atlas_labels=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc
++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/nl.xfm
+ output_dir=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain
+ AT_lin_xfm=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin.xfm
+ TA_lin_xfm=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin_inverse.xfm
+ TA_nl_xfm=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/TAnl
+ AT_nl_xfm=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/TAnl1_inverse_NL.xfm
+ atlas_label_mask=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc
+ atlas_label_mask_close=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask_fine.mnc
+ atlas_res_label_mask=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmasklinres.mnc
+++ PrintHeader /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc 1
++ python -c 'print(min([abs(x) for x in [float(x) for x in "1x1x1".split("x")]]))'
+ fixed_minimum_resolution=1.0
+++ PrintHeader /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc 1
+++ PrintHeader /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc 2
++ python -c 'print(max([ a*b for a,b in zip([abs(x) for x in [float(x) for x in "1x1x1".split("x")]],[abs(x) for x in [float(x) for x in "192x256x170".split("x")]])]))'
+ fixed_maximum_resolution=256.0
+ [[ -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc ]]
+ [[ ! -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc ]]
+++ PrintHeader /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc 1
++ python -c 'print(min([abs(x) for x in [float(x) for x in "0.3x0.3x0.3".split("x")]]))'
+ moving_minimum_resolution=0.3
++ awk -vORS= '{print "D"}'
+++ calc 'int(3/0.3+0.5)'
+++ awk 'BEGIN { print int(3/0.3+0.5) }'
++ seq 10
+ mincmorph -3D26 -clobber -successive 'B[0.5:inf:1:0]DDDDDDDDDD' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc
++ awk -vORS= '{print "D"}'
+++ calc 'int(1.5/0.3+0.5)'
+++ awk 'BEGIN { print int(1.5/0.3+0.5) }'
++ seq 5
+ mincmorph -3D26 -clobber -successive 'B[0.5:inf:1:0]DDDDD' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask_fine.mnc
+ fixedmask=NOMASK
+ movingmask=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc
+ fixedmaskfine=NOMASK
+ movingmaskfine=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask_fine.mnc
+ [[ ! -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin.xfm ]]
+ [[ ! -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin_inverse.xfm ]]
+ fixedfile=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc
+ movingfile=/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc
++ mb_ants_generate_iterations.py --min 1.0 --max 256.0 --output multilevel-halving
+ affinesteps='--transform Rigid[ 0.1 ] \
	--metric Mattes[ ${fixedfile},${movingfile},1,32,None ] \
	--convergence [ 2025x2025x2025x2025x2025x675x225x75,1e-6,10 ] \
	--shrink-factors 8x8x7x6x5x4x3x2 \
	--smoothing-sigmas 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058mm \
	--masks [ NOMASK,NOMASK ] \
--transform Similarity[ 0.1 ] \
	--metric Mattes[ ${fixedfile},${movingfile},1,64,None ] \
	--convergence [ 675x225x75,1e-6,10 ] \
	--shrink-factors 4x3x2 \
	--smoothing-sigmas 3.39728720115x2.54796540086x1.69864360058mm \
	--masks [ NOMASK,NOMASK ] \
--transform Similarity[ 0.1 ] \
	--metric Mattes[ ${fixedfile},${movingfile},1,64,None ] \
	--metric GC[ ${fixedfile},${movingfile},1,64,None ] \
	--convergence [ 675x225x75,1e-6,10 ] \
	--shrink-factors 4x3x2 \
	--smoothing-sigmas 3.39728720115x2.54796540086x1.69864360058mm \
	--masks [ ${fixedmask},${movingmask} ] \
--transform Affine[ 0.1 ] \
	--metric Mattes[ ${fixedfile},${movingfile},1,128,None ] \
	--metric GC[ ${fixedfile},${movingfile},1,128,None ] \
	--convergence [ 75x25x25,1e-6,10 ] \
	--shrink-factors 2x1x1 \
	--smoothing-sigmas 1.69864360058x0.849321800288x0mm \
	--masks [ ${fixedmaskfine},${movingmaskfine} ] '
++ eval echo --transform 'Rigid[' 0.1 ']' '\' --metric 'Mattes[' '${fixedfile},${movingfile},1,32,None' ']' '\' --convergence '[' 2025x2025x2025x2025x2025x675x225x75,1e-6,10 ']' '\' --shrink-factors 8x8x7x6x5x4x3x2 '\' --smoothing-sigmas 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058mm '\' --masks '[' NOMASK,NOMASK ']' '\' --transform 'Similarity[' 0.1 ']' '\' --metric 'Mattes[' '${fixedfile},${movingfile},1,64,None' ']' '\' --convergence '[' 675x225x75,1e-6,10 ']' '\' --shrink-factors 4x3x2 '\' --smoothing-sigmas 3.39728720115x2.54796540086x1.69864360058mm '\' --masks '[' NOMASK,NOMASK ']' '\' --transform 'Similarity[' 0.1 ']' '\' --metric 'Mattes[' '${fixedfile},${movingfile},1,64,None' ']' '\' --metric 'GC[' '${fixedfile},${movingfile},1,64,None' ']' '\' --convergence '[' 675x225x75,1e-6,10 ']' '\' --shrink-factors 4x3x2 '\' --smoothing-sigmas 3.39728720115x2.54796540086x1.69864360058mm '\' --masks '[' '${fixedmask},${movingmask}' ']' '\' --transform 'Affine[' 0.1 ']' '\' --metric 'Mattes[' '${fixedfile},${movingfile},1,128,None' ']' '\' --metric 'GC[' '${fixedfile},${movingfile},1,128,None' ']' '\' --convergence '[' 75x25x25,1e-6,10 ']' '\' --shrink-factors 2x1x1 '\' --smoothing-sigmas 1.69864360058x0.849321800288x0mm '\' --masks '[' '${fixedmaskfine},${movingmaskfine}' ']'
+++ echo --transform 'Rigid[' 0.1 ']' ' --metric' 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,32,None ']' ' --convergence' '[' 2025x2025x2025x2025x2025x675x225x75,1e-6,10 ']' ' --shrink-factors' 8x8x7x6x5x4x3x2 ' --smoothing-sigmas' 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058mm ' --masks' '[' NOMASK,NOMASK ']' ' --transform' 'Similarity[' 0.1 ']' ' --metric' 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,64,None ']' ' --convergence' '[' 675x225x75,1e-6,10 ']' ' --shrink-factors' 4x3x2 ' --smoothing-sigmas' 3.39728720115x2.54796540086x1.69864360058mm ' --masks' '[' NOMASK,NOMASK ']' ' --transform' 'Similarity[' 0.1 ']' ' --metric' 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,64,None ']' ' --metric' 'GC[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,64,None ']' ' --convergence' '[' 675x225x75,1e-6,10 ']' ' --shrink-factors' 4x3x2 ' --smoothing-sigmas' 3.39728720115x2.54796540086x1.69864360058mm ' --masks' '[' NOMASK,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc ']' ' --transform' 'Affine[' 0.1 ']' ' --metric' 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,128,None ']' ' --metric' 'GC[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,128,None ']' ' --convergence' '[' 75x25x25,1e-6,10 ']' ' --shrink-factors' 2x1x1 ' --smoothing-sigmas' 1.69864360058x0.849321800288x0mm ' --masks' '[' NOMASK,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask_fine.mnc ']'
+ antsRegistration --dimensionality 3 --verbose --minc --output '[' /tmp/tmp.CyxYrpgl3K/reg ']' --use-histogram-matching 1 --initial-moving-transform '[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1 ']' --transform 'Rigid[' 0.1 ']' --metric 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,32,None ']' --convergence '[' 2025x2025x2025x2025x2025x675x225x75,1e-6,10 ']' --shrink-factors 8x8x7x6x5x4x3x2 --smoothing-sigmas 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058mm --masks '[' NOMASK,NOMASK ']' --transform 'Similarity[' 0.1 ']' --metric 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,64,None ']' --convergence '[' 675x225x75,1e-6,10 ']' --shrink-factors 4x3x2 --smoothing-sigmas 3.39728720115x2.54796540086x1.69864360058mm --masks '[' NOMASK,NOMASK ']' --transform 'Similarity[' 0.1 ']' --metric 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,64,None ']' --metric 'GC[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,64,None ']' --convergence '[' 675x225x75,1e-6,10 ']' --shrink-factors 4x3x2 --smoothing-sigmas 3.39728720115x2.54796540086x1.69864360058mm --masks '[' NOMASK,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc ']' --transform 'Affine[' 0.1 ']' --metric 'Mattes[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,128,None ']' --metric 'GC[' /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,128,None ']' --convergence '[' 75x25x25,1e-6,10 ']' --shrink-factors 2x1x1 --smoothing-sigmas 1.69864360058x0.849321800288x0mm --masks '[' NOMASK,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask_fine.mnc ']'
 file NOMASK does not exist . 
 file NOMASK does not exist . 
 file NOMASK does not exist . 
 file NOMASK does not exist . 
 file NOMASK does not exist . 
 file NOMASK does not exist . 
+ mv -f /tmp/tmp.CyxYrpgl3K/reg0_GenericAffine.xfm /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin_inverse.xfm
+ xfminvert -clobber /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin_inverse.xfm /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin.xfm
+ [[ -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc ]]
+ [[ ! -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmasklinres.mnc ]]
+ mincresample -clobber -unsigned -byte -keep -near -like /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc -transform /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin.xfm /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmasklinres.mnc
Transforming slices:..........................................................................................................................................................................Done
+ [[ -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/labels/brain1_t1_labels.mnc ]]
++++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain
+++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1
++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations
+ mkdir -p /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/templatemasks/07ELAN15_T0_T1w_biascorr_brain
++++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain
+++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1
++ dirname /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations
+ cp /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmasklinres.mnc /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/templatemasks/07ELAN15_T0_T1w_biascorr_brain/brain1_t1.mnc
+ [[ ! -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/TAnl1_NL.xfm ]]
++ mb_ants_generate_iterations.py --min 1.0 --max 256.0
+ nonlinearsteps='--convergence [ 2025x2025x2025x2025x2025x675x225x75x25x25,1e-6,10 ] \
--shrink-factors 8x8x7x6x5x4x3x2x1x1 \
--smoothing-sigmas 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058x0.849321800288x0mm '
++ eval echo --convergence '[' 2025x2025x2025x2025x2025x675x225x75x25x25,1e-6,10 ']' '\' --shrink-factors 8x8x7x6x5x4x3x2x1x1 '\' --smoothing-sigmas 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058x0.849321800288x0mm
+++ echo --convergence '[' 2025x2025x2025x2025x2025x675x225x75x25x25,1e-6,10 ']' ' --shrink-factors' 8x8x7x6x5x4x3x2x1x1 ' --smoothing-sigmas' 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058x0.849321800288x0mm
+ antsRegistration -d 3 --collapse-output-transforms 1 --verbose --minc --initial-moving-transform /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin_inverse.xfm --transform 'SyN[0.25]' --metric 'CC[/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/templates/brains/07ELAN15_T0_T1w_biascorr_brain.mnc,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/input/atlases/brains/brain1_t1.mnc,1,4]' --use-histogram-matching 1 --output /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/TAnl --convergence '[' 2025x2025x2025x2025x2025x675x225x75x25x25,1e-6,10 ']' --shrink-factors 8x8x7x6x5x4x3x2x1x1 --smoothing-sigmas 7.64389620259x6.7945744023x5.94525260202x5.09593080173x4.24660900144x3.39728720115x2.54796540086x1.69864360058x0.849321800288x0mm --masks '[NOMASK,/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/labelmask.mnc]'
 file NOMASK does not exist . 
+ [[ ! -s /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/nl.xfm ]]
+ xfmconcat -clobber /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/ATlin.xfm /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/TAnl1_inverse_NL.xfm /home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain/output/registrations/brain1_t1/07ELAN15_T0_T1w_biascorr_brain/nl.xfm
+ rm -rf /tmp/tmp.CyxYrpgl3K
root@65f03fc9352a:/home/fmrilab/Dokumente/Studien/RUNSTUD/MagetBrain# 
After this the command stopped automatically without showing any error. 

### 

> Therefore, what is likely is, you haven't run the full pipeline.


I ran the mentioned command mb run -q parallel 

Is there any other way of running the whole pipeline?

Please let me know.
Best 
Neeraj

**Comment by gdevenyi** [comment]
>                     "You will need to run the 'voting' stage separately.")

`mb run -q parallel voting`

**Comment by neer86upadhyay** [comment]
Dear gdevenyi, 
 The command ran with suggestion you provided but showing the following error: I am providing the history of all commands we ran. 
docker run -it -v `pwd`:`pwd` -v '/home/fmrilab/Dokumente/Studien/RUNSTUD/' -w `pwd` gdevenyi/magetdocker /bin/bash
root@ea79f7b527f3:/media/fmrilab/Storage1/Habenula/MagetBrain# mb status
5 atlases, 21 templates, 111 subjects found
0 atlas-to-template registration commands left
1 template-to-subject registration commands left
11655 transform merging commands left
11655 label propagation commands left
111 label fusion commands left
root@ea79f7b527f3:/media/fmrilab/Storage1/Habenula/MagetBrain# mb run -q parallel vote -j 18 --processes 36
5 atlases, 21 templates, 111 subjects found
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

++ mktemp -d
+ tmpdir=/tmp/tmp.0qe6Zcrpfl
+ atlas=/media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/brains/05ELUS17_T0_T1w_biascorr_brain.mnc
+ target=/media/fmrilab/Storage1/Habenula/MagetBrain/input/subjects/brains/06ELCH20_T4_T1w_biascorr_brain.mnc
+ output_xfm=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/nl.xfm
++ basename /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/brains/05ELUS17_T0_T1w_biascorr_brain.mnc .mnc
+ atlas_stem=05ELUS17_T0_T1w_biascorr_brain
++ basename /media/fmrilab/Storage1/Habenula/MagetBrain/input/subjects/brains/06ELCH20_T4_T1w_biascorr_brain.mnc .mnc
+ target_stem=06ELCH20_T4_T1w_biascorr_brain
+++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/brains/05ELUS17_T0_T1w_biascorr_brain.mnc
++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/brains
+ atlas_labels=/media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/labels/05ELUS17_T0_T1w_biascorr_brain_labels.mnc
++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/nl.xfm
+ output_dir=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain
+ AT_lin_xfm=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/ATlin.xfm
+ TA_lin_xfm=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/ATlin_inverse.xfm
+ TA_nl_xfm=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl
+ AT_nl_xfm=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl1_inverse_NL.xfm
+ atlas_label_mask=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmask.mnc
+ atlas_label_mask_close=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmask_fine.mnc
+ atlas_res_label_mask=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmasklinres.mnc
+++ PrintHeader /media/fmrilab/Storage1/Habenula/MagetBrain/input/subjects/brains/06ELCH20_T4_T1w_biascorr_brain.mnc 1
++ python -c 'print(min([abs(x) for x in [float(x) for x in "1x1x1".split("x")]]))'
+ fixed_minimum_resolution=1.0
+++ PrintHeader /media/fmrilab/Storage1/Habenula/MagetBrain/input/subjects/brains/06ELCH20_T4_T1w_biascorr_brain.mnc 1
+++ PrintHeader /media/fmrilab/Storage1/Habenula/MagetBrain/input/subjects/brains/06ELCH20_T4_T1w_biascorr_brain.mnc 2
++ python -c 'print(max([ a*b for a,b in zip([abs(x) for x in [float(x) for x in "1x1x1".split("x")]],[abs(x) for x in [float(x) for x in "192x256x170".split("x")]])]))'
+ fixed_maximum_resolution=256.0
+ [[ -s /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/labels/05ELUS17_T0_T1w_biascorr_brain_labels.mnc ]]
+++++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/brains/05ELUS17_T0_T1w_biascorr_brain.mnc
++++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates/brains
+++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/input/templates
++ dirname /media/fmrilab/Storage1/Habenula/MagetBrain/input
+ [[ -d /media/fmrilab/Storage1/Habenula/MagetBrain/output/templatemasks/05ELUS17_T0_T1w_biascorr_brain ]]
+ [[ ! -s /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmask.mnc ]]
+ fixedmask=NOMASK
+ fixedmaskfine=NOMASK
+ movingmask=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmask.mnc
+ movingmaskfine=/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmask_fine.mnc
+ [[ ! -s /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/ATlin.xfm ]]
+ [[ -s /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmask.mnc ]]
+ [[ ! -s /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/labelmasklinres.mnc ]]
+ [[ ! -s /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl1_NL.xfm ]]
+ [[ ! -s /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/nl.xfm ]]
+ xfmconcat -clobber /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/ATlin.xfm /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl1_inverse_NL.xfm /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/nl.xfm
/home/nistmni/src/minc-toolkit-v2/libminc/volume_io/Prog_utils/print.c:226 (from MINC): volume_io error: Error:  could not open file "/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl1_inverse_NL.xfm".  
/home/nistmni/src/minc-toolkit-v2/libminc/volume_io/Prog_utils/print.c:226 (from MINC): volume_io error: 
System message: No such file or directory

xfmconcat: Error reading transform file /media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl1_inverse_NL.xfm
Traceback (most recent call last):
  File "/opt/MAGeTbrain/bin/mb", line 792, in <module>
    main()
  File "/opt/MAGeTbrain/bin/mb", line 220, in main
    options.func(options)
  File "/opt/MAGeTbrain/bin/mb", line 329, in command_run
    p.run()
  File "/opt/MAGeTbrain/bin/mb", line 608, in run
    self.queue.run(stages=self._get_stage_order())
  File "/opt/MAGeTbrain/bin/mb", line 508, in run
    self.parallel(self.commands[stage])
  File "/opt/MAGeTbrain/bin/mb", line 514, in parallel
    self.execute(command, input='\n'.join(commands))
  File "/opt/MAGeTbrain/bin/mb", line 492, in execute
    raise Exception("Returns %i :: %s" % (proc.returncode, command))
Exception: Returns 1 :: parallel -j36


We are also not able to find this file:

xfmconcat: Error reading transform file **/media/fmrilab/Storage1/Habenula/MagetBrain/output/intermediate/05ELUS17_T0_T1w_biascorr_brain/06ELCH20_T4_T1w_biascorr_brain/TAnl1_inverse_NL.xfm**

How can we create it?
Please have a look and help us in sorting this issue.

Best Regards,

**Comment by gdevenyi** [comment]
```
1 template-to-subject registration commands left
```

Only a single registration command is left and failing.

Something about that subject is different than the others
1. Too large.
2. Too high resolution
3. Broken image.

**Comment by neer86upadhyay** [comment]
Dear Gabriel, 
Thanks for your suggestion. We were able to ran the whole pipeline and finally extracted the Habenula volumes but unfortunately the range of volume (left:6-22; right 9-20 mm3) for 3T MRI data is very different from what is already mentioned in the literature. 
Could you provide your comments and suggestions what may have possibly affected the segmentation?

Best Wishes,
Neeraj

**Comment by gdevenyi** [comment]
Your volumes are within the reported range from our publication. https://ars.els-cdn.com/content/image/1-s2.0-S2451902220300227-mmc1.pdf

However, you have made no mention of attempting any quality control. The ultimate determination of correctness for a segmentation is that machine-learning that is the human visual system comparing the segmentations to the underlying anatomy.

Other factors that impact Hb segmentation
1. being small, voxel size has a large impact
2. population
3. scan parameter choices affecting contrast

As we have addressed all the "MAGeTdocker doesn't work" you've reported, I'm closing this issue. Please open an new issue if you have new topics/questions to discuss.
